.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,51 @@
+<!-- 
+Please use this template when submitting a new feature request with as much details as possible. Not doing so may result in the issue not being addressed in a timely manner or eventually closed.
+-->
+# Title
+<!-- 
+The same title used to create the issue (optional)
+-->
+
+## Summary 
+
+<!-- 
+A quick summary about the feature request or bug fix with issue number. Ex: Add a new logging system described in #<issue number> 
+-->
+
+## Type
+
+[] Bug fix
+[] Feature request
```

Comment:
Should a feature request not be opened as an issue? Or are we planning to keep feature proposals somewhere in the repo before implementation?

Reply:
@JoelSpeed the proposal here is to be able to differentiate which pull requests are related to feature requests or bug fixes. So we can set the proper label and filter it.

This is how I see it:
- Feature requests: new features that people would like to see on Louketo. For example: Implement support to multiple upstreams.
- Enhancement: things like a typo, a small refactoring into the code. 

Maybe we can get rid of these labels and pick just one.


For feature proposals, something that will be a large feature and will change drastically the codebase, we can use [louketo-community](https://github.com/louketo/louketo-community) and create a folder "design-proposals". Does it make any sense?

But this is just an idea, we can change it, absolutely nothing is set in stone. Including the current labels, which I created to quickly migrate all the Jiras.

How does it work actually on the oauth2-proxy repository? If there's anything that you think should be changed, please let me know and we can adjust as we go.

Reply:
That sounds perfectly reasonable to me, happy to keep this in. We can always iterate if we need to later.

Currently for categorising, normally the reviewer will look at it and mark it as the correct type. For proposals, we prefer to have new features discussed in an issue but this isn't always done and we haven't been strict about requiring it in the past. We actually discussed recently about changing the templates to tell people to create an issue to discuss changes before implementing them.